runtime/lib_ksils12.c

```diff
@@ -409,6 +414,82 @@ tlvCreateMetadata(ksifile ksi, uint64_t record_index, const char *key,
 	return r;
 }
 
+#define KSI_FILE_AMOUNT_INC 32
+
+static int
+rsksiRegisterKsiFile(rsksictx ctx, ksifile ksi) {
+	int ret = RSGTE_INTERNAL;
+
+	if(ctx == NULL || ksi == NULL) {
+		return RSGTE_INTERNAL;
+	}
+
+	/* To be extra sure that ksifile buffer is initialized correctly, clear variables. */
+	if (ctx->ksi == NULL) {
+		ctx->ksiCount = 0;
+		ctx->ksiCapacity = 0;
+		ctx->ksiCapacity = 0;
+	}
+
+	/* If needed allocate memory for the buffer. */
+	if (ctx->ksiCount == ctx->ksiCapacity) {
+		size_t i = 0;
+		ksifile *tmp = NULL;
+
+		tmp = (ksifile*)realloc(ctx->ksi,  sizeof(ksifile) * (ctx->ksiCapacity + KSI_FILE_AMOUNT_INC));
+		if (tmp == NULL) {
+			ret = RSGTE_OOM;
+			goto done;
+		}
+
+		for (i = 0; i < KSI_FILE_AMOUNT_INC; i++) {
+			tmp[ctx->ksiCapacity + i] = NULL;
+		}
+
+		ctx->ksi = tmp;
+		tmp = NULL;
+
+		ctx->ksiCapacity += KSI_FILE_AMOUNT_INC;
+	}
```

Comment:
move this block to a separate function

Reply:
New function `rsksiExpandRegisterIfNeeded` added.